features/basic.feature

```diff
@@ -1,16 +1,14 @@
-Uses: StepRegistry
+Uses: Web Sockets
 Uses: Selenium
 Uses: Chorus Context
 
 Feature: Basic steps
 
-  #! StepRegistry start
   Feature-Start:
-    Given I open Chrome
-
-  Background:
-    Given I navigate to http://localhost:9999
-    And StepRegistry client SimpleStepPublisher is connected
+    Given I start a web socket server
+    And I open Chrome
+    And I navigate to http://localhost:9999
```

Comment:
Just checking here, but putting this in the start section, doesn't that mean that you are refreshing the page only once at the beginning? I used to have it in the background block so that each test doesn't depend on previous state.

Reply:
Chorus features often involve some heavy lifting, such as starting processes and making connections at the start of the feature. Since this can be time consuming, a common pattern is to put the initial setup steps into the Feature-Start: section, and use a quicker 'reset' step at the start of each scenario. As the test suite grows this saves substantial time. In this case, since the 'navigate' step is relatively fast it probably wouldn't make much difference (although that step does trigger the web socket to reconnect as the page loads). So we could do either for these tests, although arguably the 'reset' captures the intent of resetting the state better than the 'navigate'

Reply:
Agreed!

Reply:
Thanks for the clarification.